### ISSUE TITLE
[FEATURE] Allow to go to homepage home in x env ll

### ISSUE THREAD

**Issue by iandol** [x-cmd/x-cmd#38]
**Describe the feature**

For `x env --app` or `x elv ll` there is pkg info, including the homepage. It would be great if we could press <kbd>h</kbd> and the homepage would open.

```
╭───────────────────┬───────────────────┬───────────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ai                 │asciinema          │v1.24.0            │size: Unknown                                                                                            │
 │build              │bat                │                   │desc: just 是保存和运行项目特定命令的便捷方法                                                            │
 │cicd               │buku               │                   │homepage: https://github.com/casey/just                                                                  │
 │cloud              │busybox            │                   │bin: just                                                                                                │
```

Another small issue, when you scroll down the pkg list the 4th pane (the one that shows the package details is not shown, so it is hard to know what the package does. It would be great if when scrolling down the 2nd pane (package name), the default version and pkg info was visible. If you want to select another version then tab and select it, but it shouldn't be necessary to focus the version to see the package information:

```
╭───────────────────┬───────────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ai                 │7za                │v23.01                                                                                                                       │
 │build              │7zz                │v23.01.00                                                                                                                    │
 │cicd               │archiver           │                                                                                                                             │
 │cloud              │brook              │                                                                                                                             │
 │common             │csview             │                                                                                                                             │
 │container          │csvkit             │                                                                                                                             │
 │data               │dasel              │                                                                                                                             │
 │db                 │decktape           │                                                                                                                             │
 │editor             │deepl              │                                                                                                                             │
 │filesystem         │diffoscope         │                                                                                                                             │
 │git                │difft              │                                                                                                                             │
 │iac                │docsify-cli        │                                                                                                                             │
 │js                 │dvc                │                                                                                                                             │
 │lang               │dyff               │                                                                                                                             │
 │lint               │eslint             │                                                                                                                             │
 │linter             │fanyi              │                                                                                                                             │
 │multimedia         │fq                 │                                                                                                                             │
 │network            │franc-cli          │                                                                                                                             │
 │osman              │fx                 │                                                                                                                             │
 │other              │gojo               │                                                                                                                             │
 │package-manager    │gojq               │                                                                                                                             │
 │plugin             │gomplate           │                                                                                                                             │
 │python             │grex               │                                                                                                                             │
 │reader             │gron               │                                                                                                                             │
 │sdkman             │hcl2json           │                                                                                                                             │
 ╰───────────────────┴───────────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
 CANDIDATE: 7zz                                                                                                                                                         
```

**Is is urgent**
No

**Comment by edwinjhlee** [comment]
Nice ~ We will add more options in the release (v0.3.15) next week.

**Comment by edwinjhlee** [comment]
Thank you for your feedback about the TUI. There's room for improvement in x-cmd's TUI application.

I am not satisfied with our current TUI framework ( although this is our forth version and is considered as a break through from the last version ) ... I would develop a new TUI framework, after that, we will develop module-specific TUI applications with a consistent user experience in the future. 


Currently, our focus is on the AI, Hub, and PKG systems. We aim to integrate these components into a cohesive ecosystem. 
    
Once this integration is complete, we plan to promote x-cmd, and at the same time, take a break ~~~


As always, thanks for your support.

**Comment by edwinjhlee** [comment]
> **Describe the feature**
> 
> For `x env --app` or `x elv ll` there is pkg info, including the homepage. It would be great if we could press h and the homepage would open.
> 
> ```
>  ╭───────────────────┬───────────────────┬───────────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────╮
>  │ai                 │asciinema          │v1.24.0            │size: Unknown                                                                                            │
>  │build              │bat                │                   │desc: just 是保存和运行项目特定命令的便捷方法                                                            │
>  │cicd               │buku               │                   │homepage: https://github.com/casey/just                                                                  │
>  │cloud              │busybox            │                   │bin: just                                                                                                │
> ```
> 
> Another small issue, when you scroll down the pkg list the 4th pane (the one that shows the package details is not shown, so it is hard to know what the package does. It would be great if when scrolling down the 2nd pane (package name), the default version and pkg info was visible. If you want to select another version then tab and select it, but it shouldn't be necessary to focus the version to see the package information:
> 
> ```
>  ╭───────────────────┬───────────────────┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
>  │ai                 │7za                │v23.01                                                                                                                       │
>  │build              │7zz                │v23.01.00                                                                                                                    │
>  │cicd               │archiver           │                                                                                                                             │
>  │cloud              │brook              │                                                                                                                             │
>  │common             │csview             │                                                                                                                             │
>  │container          │csvkit             │                                                                                                                             │
>  │data               │dasel              │                                                                                                                             │
>  │db                 │decktape           │                                                                                                                             │
>  │editor             │deepl              │                                                                                                                             │
>  │filesystem         │diffoscope         │                                                                                                                             │
>  │git                │difft              │                                                                                                                             │
>  │iac                │docsify-cli        │                                                                                                                             │
>  │js                 │dvc                │                                                                                                                             │
>  │lang               │dyff               │                                                                                                                             │
>  │lint               │eslint             │                                                                                                                             │
>  │linter             │fanyi              │                                                                                                                             │
>  │multimedia         │fq                 │                                                                                                                             │
>  │network            │franc-cli          │                                                                                                                             │
>  │osman              │fx                 │                                                                                                                             │
>  │other              │gojo               │                                                                                                                             │
>  │package-manager    │gojq               │                                                                                                                             │
>  │plugin             │gomplate           │                                                                                                                             │
>  │python             │grex               │                                                                                                                             │
>  │reader             │gron               │                                                                                                                             │
>  │sdkman             │hcl2json           │                                                                                                                             │
>  ╰───────────────────┴───────────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
>  CANDIDATE: 7zz                                                                                                                                                         
> ```
> 
> **Is is urgent** No

Hi Ian,

I'm excited to invite you to participate in the insider preview of x7! This new release will allow you to receive daily updates for x-cmd.

Please be aware that x7 is a preview release and may contain bugs or unfinished features. I'm currently preparing documentation outlining potential risks and known issues.

Would you be interested in joining the x7 insider preview and receiving daily x-cmd updates?

**Comment by iandol** [comment]
Yes, that sounds great, I am happy to preview test, accepting there may be instability etc.

**Comment by iandol** [comment]
Should I will close this, assuming once the new TUI comes in we can then assess what features can be implemented. There are several places where I would like a bit more functionality from the TUI interfaces, but it doesn't make sense until we have the new TUI ...

**Comment by edwinjhlee** [comment]
> Should I will close this, assuming once the new TUI comes in we can then assess what features can be implemented. There are several places where I would like a bit more functionality from the TUI interfaces, but it doesn't make sense until we have the new TUI ...

    This issue can remain open. 
    
    There are several outstanding tasks. The TUI framework may be rewritten in the future, potentially using Zig.  
    
    For now, I will prioritize ensuring the stability of x-cmd and focus on AI and hub development. X-CMD is much better with AI. But our current version fail to show that picture. Shame.
